mapsims/tests/test_noise_extern.py

```diff
@@ -0,0 +1,193 @@
+
+    def get_hitsmaps(self):
+        return self.hitsmaps
+    
+    def get_hitsmap_filenames(self):
+        return self.hitsmap_filenames
+    
+    def get_ivar_maps(self):
+        return self.ivar_maps
+
+    def get_ivar_map_filenames(self):
+        return self.ivar_map_filenames
+
+    def get_white_noise(self,f_sky, units='arcmin2',*args,**kwargs):
+        if self.white_noises is None:
+            return None
+        A = self.white_noises**2
+        if units == 'arcmin2':
+            A *= (60*180/np.pi)**2
+        elif units != 'sr':
+            raise ValueError("Unknown units '%s'." % units)
+        return A
+    
+    def get_noise_curves(self, f_sky, ell_max, delta_ell, deconv_beam=True,
+                         full_covar=False, rolloff_ell=None):
+        ell = np.arange(2, ell_max, delta_ell)
+        T_out = np.zeros([self.nbands,self.nbands,ell.shape[0]])
+        P_out = np.zeros([self.nbands,self.nbands,ell.shape[0]])
+        if self.noise_ell is not None:
+            if len(np.shape(self.noise_TT))==3:
+                for i in range(self.nbands):
+                    for j in range(self.nbands):
+                        # noise_ells,noise_T,noise_P = np.loadtxt(self.noise_files[i][j],unpack=True)
+                        T_out[i,j] = interp1d(self.noise_ell,self.noise_TT[i,j],bounds_error=False,fill_value=0)(ell)
+                        P_out[i,j] = interp1d(self.noise_ell,self.noise_PP[i,j],bounds_error=False,fill_value=0)(ell)    
+            else:
+                for i in range(self.nbands):
+                    T_out[i,i] = interp1d(self.noise_ell,self.noise_TT[i],bounds_error=False,fill_value=0)(ell)
+                    P_out[i,i] = interp1d(self.noise_ell,self.noise_PP[i],bounds_error=False,fill_value=0)(ell)
+        else:
+            if np.shape(self.noise_files)==2:
+                for i in range(self.nbands):
+                    for j in range(self.nbands):
+                        noise_ell,noise_TT,noise_PP = np.loadtxt(self.noise_files[i][j],unpack=True)
+                        T_out[i,j] = interp1d(noise_ell,noise_TT,bounds_error=False,fill_value=0)(ell)
+                        P_out[i,j] = interp1d(noise_ell,noise_TP,bounds_error=False,fill_value=0)(ell)
+                
+            else:
+                for i in range(self.nbands):
+                    noise_ell,noise_TT,noise_PP = np.loadtxt(self.noise_files[i],unpack=True)
+                    T_out[i,i] = interp1d(noise_ell,noise_TT,bounds_error=False,fill_value=0)(ell)
+                    P_out[i,i] = interp1d(noise_ell,noise_PP,bounds_error=False,fill_value=0)(ell)
+        return (ell, T_out, P_out)
+
+
+if __name__=='__main)_':
+    
+    # Test that this code and the SONoiseSimulator agree if the above used the same noise curves and hitsmap.
+    nside = 16
+    SONoiseSimulator = mapsims.noise.SONoiseSimulator(nside=nside)
+
+    ell, noise_ell_T, noise_ell_P = SONoiseSimulator._get_survey('LT3').get_noise_curves(
```

Comment:
I think `get_survey` should not have an underscore, it seems pretty useful to use externally